spacy_experimental/span_boundary_detection/sbd_component.py

```diff
@@ -0,0 +1,301 @@
+from typing import List, Dict, Callable, Tuple, Optional, Iterable, Any
+from thinc.api import Config, Model, set_dropout_rate
+from thinc.api import Optimizer
+from thinc.types import Floats2d
+from numpy import float32
+
+from spacy.language import Language
+from spacy.util import registry
+from spacy.pipeline.trainable_pipe import TrainablePipe
+from spacy.tokens import Doc, Token
+from spacy.training import Example
+from spacy.scorer import PRFScore
+
+SBD_DEFAULT_CONFIG = """
+[model]
+@architectures = "spacy-experimental.span_boundary_detection_model.v1"
+
+[model.scorer]
+@layers = "spacy.LinearLogistic.v1"
+nO = 2
+
+[model.tok2vec]
+@architectures = "spacy.Tok2Vec.v1"
+
+[model.tok2vec.embed]
+@architectures = "spacy.MultiHashEmbed.v1"
+width = 96
+rows = [5000, 2000, 1000, 1000]
+attrs = ["ORTH", "PREFIX", "SUFFIX", "SHAPE"]
+include_static_vectors = false
+
+[model.tok2vec.encode]
+@architectures = "spacy.MaxoutWindowEncoder.v1"
+width = ${model.tok2vec.embed.width}
+window_size = 1
+maxout_pieces = 3
+depth = 4
+"""
+
+DEFAULT_SBD_MODEL = Config().from_str(SBD_DEFAULT_CONFIG)["model"]
+
+
+@Language.factory(
+    "spacy-experimental_span_boundary_detection_component_v1",
```

Comment:
May I suggest: `experimental_span_boundary_detector`?

The factory names usually don't include `spacy` or version numbers.